pkg/metrics/instance/instance_integration_test.go

```diff
@@ -73,9 +73,14 @@ remote_write: []
 	require.NoError(t, err)
 
 	instCtx, cancel := context.WithCancel(context.Background())
-	defer cancel()
+	done := make(chan struct{})
+	defer func() {
+		cancel()
+		<-done
+	}()
```

Comment:
A slightly more idiomatic way of doing this is with [sync.WaitGroup](https://pkg.go.dev/sync#WaitGroup):

```go 
var wg sync.WaitGroup
defer func() {
  cancel() 
  wg.Wait() 
}

wg.Add(1) // Do this before launching the goroutine 
go func() {
  defer wg.Done() 
  // business logic here 
}
```

It tends to be slightly more code than what you have (because you have to explicitly add deltas to the waitgroup), but it doesn't have the risk of closing the same channel twice, and it works across multiple goroutines.

What you have now is perfectly fine, but I just wanted to give you a heads up about sync.WaitGroup :)

Reply:
Thanks, I made the change :)